pandas/core/tools/datetimes.py

```diff
@@ -763,7 +763,7 @@ def to_datetime(
         Return type depends on input:
 
         - list-like:
-            - DatetimeIndex, if timezone naive or aware with the same timezone
+            - DatetimeIndex, if timezone naive or aware with constant time offset
```

Comment:
how about leave the original & add this as well

Reply:
I am not sure to understand your comment @jreback , sorry about this. Could you be more precise please ? Do you mean a sentence such as 

```suggestion
            - DatetimeIndex, if timezone naive or timezone-aware with either
              constant time offset or constant timezone name.
```

?

Reply:
I get what you're trying to convey here (a timezone can have different UTC offsets), but I think the current language is fine.

I think a better contribution here would be to add a `Notes` section referencing pandas timezone handling so users can learn more about these subtleties. https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#time-zone-handling

Reply:
Thanks for the suggestion @mroeschke ! Sorry for not having had the bandwidth to respond earlier.

The current docstring opposes two situations:

 - DatetimeIndex \[are returned\] if \[input datetimes are\] timezone naive or aware with the same timezone
 - Index of object dtype \[are returned\] if \[input datetimes are\] timezone aware with mixed time offsets

The first bullet is incorrect or at least very misleading, because it conflicts with the second bullet. Indeed datetimes that are "aware with the same timezone" **can** contain mixed time offsets. My proposal removes the conflict:

 - DatetimeIndex \[are returned\] if \[input datetimes are\] timezone naive or aware with **constant time offset**
 - Index of object dtype \[are returned\] if \[input datetimes are\] timezone aware with mixed time offsets

Does this make sense ?

Now concerning your suggestion for adding a "Notes" section, I think that it is great ! I'll include this link somewhere